### PR TITLE
[build] don't embed bitcode for target libs built for simulators

### PR DESCRIPTION
**Pull request by edymtt**
At the same time rework condition so that it is more amenable to code
searches.

Addresses rdar://65501878

### CODE REVIEW

**Comment by edymtt** [comment]
@swift-ci please smoke test

File: stdlib/cmake/modules/AddSwiftStdlib.cmake [hunk 1]
```diff
@@ -1251,7 +1251,9 @@ function(_add_swift_target_library_single target name)
     endif()
     # Include LLVM Bitcode slices for iOS, Watch OS, and Apple TV OS device libraries.
     if(SWIFT_EMBED_BITCODE_SECTION AND NOT SWIFTLIB_SINGLE_DONT_EMBED_BITCODE)
-      if(${SWIFTLIB_SINGLE_SDK} MATCHES "(I|TV|WATCH)OS")
+      if(${SWIFTLIB_SINGLE_SDK} STREQUAL "IOS" OR
+         ${SWIFTLIB_SINGLE_SDK} STREQUAL "IOS" OR
```
Comment (compnerd) [inline stdlib/cmake/modules/AddSwiftStdlib.cmake:1255]: Was this meant to be `TVOS`?

Comment (edymtt) [reply stdlib/cmake/modules/AddSwiftStdlib.cmake]: Thanks for catching this -- fixed with 09fe71f5441

**Comment by edymtt** [comment]
@swift-ci please smoke test

**Comment by edymtt** [comment]
@swift-ic please test Windows

**Comment by edymtt** [comment]
@swift-ci  please test Windows

**Comment by edymtt** [comment]
@swift-ci please smoke test macOS

**Review by compnerd — APPROVED** [review]

**Pull request merged**